service/index.js

```diff
@@ -0,0 +1,40 @@
+const Contact = require("./schemas/schemas");
+const mongoose = require("mongoose");
+
+const getAllContacts = () => {
+  return Contact.find();
+};
+
+const getContactById = (id) => {
+  if (mongoose.Types.ObjectId.isValid(id)) {
+    return Contact.findOne({ _id: id });
+  } else {
+    return null;
+  }
+};
+
+const createContact = ({ name, email, phone, favorite }) => {
+  return Contact.create({ name, email, phone, favorite });
+};
+
+const removeContact = (id) => {
+  if (mongoose.Types.ObjectId.isValid(id)) {
+    return Contact.findOneAndRemove({ _id: id });
+  }
+  return null;
+};
```

Comment:
to samo tutaj

Reply:
w tych dwoch funkcjach `getContactById` i `removeContact` masz w sumie te sama logike, zmienia sie tylko metoda z `Contact`, wiec mozesz zrobic funkcje wyzszego rzedu i wrzucac callback! :)

```js
const validateContactAnd = (cb, id) => 
  mongoose.Types.ObjectId.isValid(id) 
    ? cb({ _id: id }) 
    : null;
```

uzycie: `const removeContact = (id) => validateContactAnd(Contact.findOneAndRemove, id);`

nie musisz tego robic, ale mozesz :)